src/templates/array3D.h

```diff
@@ -86,9 +86,19 @@ template <class A> class Array3D
             resize(nX, nY, nZ);
     }
 
+    // Return const reference to values
+    const std::vector<A> &values() const { return array_; }
+
+    // Return modifiable reference to values
+    std::vector<A> &values() { return array_; }
```

Comment:
Tens to write these without a separate comment for the overload
```suggestion
    // Return reference to values
    std::vector<A> &values() { return array_; }
    const std::vector<A> &values() const { return array_; }
```